.github/workflows/ci.yaml

```diff
@@ -32,10 +32,13 @@ jobs:
           # check package is importable
           python -c "import sql"
           pip install ".[dev]"
-
+          pip install sqlalchemy -U
 
       - name: Test with pytest
         run: |
+          # ensure we're running sqlalchemy 2
+          python -c "import sqlalchemy; assert int(sqlalchemy.__version__.split('.')[0]) == 2"
+          # run unit tests
           pytest --durations-min=5 --ignore=src/tests/integration
 
   test-sqlalchemy-v1:
```

Comment:
@edublancas 
Do you know the clear context of this to answer Ido? Since I was referring to the previous PR to do this

For my understanding:
This is making sure the CI container uses sqlalchemy 2, if the assert condition is false, the test case is early failed

Reply:
we broke a previous release because the pip resolver was installing sqlalchemy 1.x even though we were doing `pip install sqlalchemy`; this is a sanity check to ensure we're actually testing against sqlalchemy 2.x

this will add some CI extra time since we're adding new jobs. but we're testing sqlalchemy 1.x only with Python 3.11. and since github runs the build jobs in parallel, it won't have much of an effect. We can decide later if we expand or reduce this. I think for now, it's fine